collection-scripts/common.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+# shellcheck disable=SC2034,SC2086,SC2001
+
+export DST_DIR="must-gather"
+
+# run must-gather in the namespaces one by one
+function run_mustgather() {
+    for ns in "$@"; do
+        oc adm inspect "${log_collection_args}" "namespace/$ns" --all-namespaces --dest-dir "$DST_DIR" || echo "Error inspecting namespace/$ns"
+    done
+}
+
+# get the list of namespaces where defined resources exist
+function get_all_namespace() {
+    local nslist
+    for kind in "$@"; do
+        nslist+=$(oc get "$kind" --all-namespaces -o=jsonpath='{range .items[*]}{.metadata.namespace}{"\n"}{end}')
+    done
+    uniq_list "$nslist"
+}
+
+# remove dupplicated namespaces
+function uniq_list() {
+    echo "$@" | sort -u
+}
+
+# get the version of rhoai operator
+function version() {
+  version=''
+
+  # get version from rhoai csv
+  csv_name=$(oc get csv|grep rhods|awk '{print $1}')
+  [[ z == z"${version}" && z != z"${csv_name}" ]] && version=$(oc get csv "$(oc get csv|grep rhods|awk '{print $1}')" -o json|jq .spec.version |xargs)
```

Comment:
Wondering why that old shell programming trick with x$VAR used here? In modern bash -z/-n and even comparation with empty string should work inside of [[ operator.

Reply:
thanks, updated in new commit

Reply:
same reason for why not update them at first.